lib/scheduler/helpers.go

```diff
@@ -0,0 +1,68 @@
+package scheduler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math"
+	"strings"
+)
+
+// A helper function to verify percentage distributions
+func checkPercentagesSum(percentages []float64) error {
+	var sum float64
+	for _, v := range percentages {
+		sum += v
+	}
+	if math.Abs(100-sum) >= minPercentage {
+		return fmt.Errorf("split percentage sum is %.2f while it should be 100", sum)
+	}
+	return nil
+}
+
+// A helper function for joining error messages into a single string
+func concatErrors(errors []error, separator string) string {
+	errStrings := make([]string, len(errors))
+	for i, e := range errors {
+		errStrings[i] = e.Error()
+	}
+	return strings.Join(errStrings, separator)
+}
+
+// Decode a JSON in a strict manner, emitting an error if there are unknown fields
+func strictJSONUnmarshal(data []byte, v interface{}) error {
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.DisallowUnknownFields()
+	dec.UseNumber()
+
+	if err := dec.Decode(&v); err != nil {
+		return err
+	}
+	if dec.More() {
+		return fmt.Errorf("unexpected data after the JSON object")
```

Comment:
Can you write a test for this ?

Reply:
Yeah, will do. Apparently, `json.Unmarshal()` sends only properly formatted json object my way, so that's why the current tests didn't catch it. I'll add a separate simple test for just `strictJSONUnmarshal` that tests it...

Reply:
Done